docs/admin/templates/extending-templates/devcontainers.md

```diff
@@ -1,126 +1,340 @@
+the code repository.
+
+When integrated with Coder templates, dev containers provide:
+
+- **Project-specific environments**: Each repository can define its own tools, extensions, and configuration.
+- **Faster start times**: Developers start workspace with fully configured environments without additional installation.
+- **Consistency across teams**: Everyone works in identical environments regardless of their local machine.
+- **Version-controlled environments**: Development environment changes are tracked alongside code changes.
+
+The Dev Container integration replaces the legacy Envbuilder integration.
+Visit [Choose an approach to Dev Containers](./dev-containers-envbuilder.md) for more information about how they compare.
+
+## Prerequisites
+
+- Dev containers require Docker to build and run containers inside the workspace.
+
+  Ensure your workspace infrastructure has Docker configured with container creation permissions and sufficient resources.
+
+  If it doesn't, follow the steps in [Docker in workspaces](./docker-in-workspaces.md).
+
+- The `devcontainers-cli` module requires npm.
+
+  - You can use an image that already includes npm, such as `codercom/enterprise-node:ubuntu`.
+
 ## Install the Dev Containers CLI
 
 Use the
 [devcontainers-cli](https://registry.coder.com/modules/devcontainers-cli) module
-to ensure the `@devcontainers/cli` is installed in your workspace:
+to install `devcontainers/cli` in your workspace:
 
 ```terraform
 module "devcontainers-cli" {
-  count    = data.coder_workspace.me.start_count
-  source   = "dev.registry.coder.com/modules/devcontainers-cli/coder"
-  agent_id = coder_agent.dev.id
+  count = data.coder_workspace.me.start_count
+  source = "registry.coder.com/modules/devcontainers-cli/coder"
+  agent_id = coder_agent.main.id
 }
 ```
 
-Alternatively, install the devcontainer CLI manually in your base image.
+Alternatively, you can install `@devcontainer/cli` manually in your base image:
 
-## Configure Automatic Dev Container Startup
+```shell
+RUN npm install -g @devcontainers/cli
+```
 
-The
-[`coder_devcontainer`](https://registry.terraform.io/providers/coder/coder/latest/docs/resources/devcontainer)
-resource automatically starts a dev container in your workspace, ensuring it's
-ready when you access the workspace:
+## Configure the Agent for Docker Support
+
+Your Coder agent needs proper configuration to support Docker and dev containers:
+
+```terraform
+resource "coder_agent" "main" {
+  os = "linux"
+  arch = "amd64"
+
+  # Ensure Docker starts before the agent considers the workspace ready
+  startup_script_behavior = "blocking"
+  startup_script = "sudo service docker start"
+
+  # Properly shut down Docker when the workspace stops
+  shutdown_script = "sudo service docker stop"
+}
+```
+
+The `blocking` behavior ensures Docker is fully started before the workspace is considered ready, which is critical for dev containers to function correctly.
+
+## Define the Dev Container Resource
```

Comment:
I need to review the order of the sections here - the heading is about defining the resource, but it starts with git-clone